view/frontend/templates/js/replacejs.phtml

```diff
@@ -806,8 +805,12 @@ ob_start();
                         createRequest = false;
                     })
             });
+            // Get number of items in cart from counter element
+            var numItems = $(".counter-number").text();
 
-            var BC = BoltCheckout.configure(cart, hintBarrier.promise, callbacks);
+            // Call config with floatingButtonMode if always present checkout enabled and items were or were not just added to cart (dataChanged)
+            var config = boltConfig.always_present_checkout && (numItems && numItems !=="0" || dataChanged)? {floatingButtonMode: dataChanged ? "show_cart" : "show_cart_on_hover" } : {};
+            var BC = BoltCheckout.configure(cart, hintBarrier.promise, callbacks, config);
```

Comment:
We have three configure call in this file. Could you please add your code to two others as well? It's possible that numItems still 0 at this moment because data is still loading. Later we re-configure bolt, but from another place and don't calculate config.
I think we can just copy-paste the code. We are planning to refactor this JS code soon.

Reply:
Updated. I only want to call with "show_cart" when a new item is added (when ajax:addToCart event is fired) so for other two places I call with only "show_cart_on_hover" and remove dataChanged param.

Reply:
Actually we can't call everywhere. The issue is that we only want to call with config when a cart with items is created (which to my knowledge happens only in `createOrder`. If I call with config later `BC.configure` is called with a null cart. This breaks always present button as it depends on having cart to show. We only want to call it when passing a cart that is not null.

Reply:
I also changed monitorChagnes function quite a bit which should make fix edge cases where number hasn't updated uet.

Reply:
On my test site when I am on product page:
1) The first configure call happens on line 813. numItem is "0" at this moment, we don't show always present button
2) The second call happens on line 916. Cart is set and we can check $(".counter-number").text() and show button if we copy the same code before line 916.

So for now on my test site button is shown if I add product to cart, but isn't shown if reload the page.